code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -261,6 +228,31 @@ meteor_act
 	damage -= (toughness * stat_affect + item_size_affect)
 	return max(0, damage)
 
+/mob/living/carbon/human/proc/grab_redirect_attack(var/obj/item/grab/G, var/obj/item/I)
+	var/mob/living/carbon/human/attacker = G.assailant
+	var/mob/living/carbon/human/grabbed = G.affecting
+	visible_message(SPAN_DANGER("[src] redirects the blow at [grabbed]!"), SPAN_DANGER("You redirect the blow at [grabbed]!"))
+	//check what we are being hit with, a hand(I is null), or an item?
+	//quickly turn blocking off and on to prevent looping(since we are attacking again)
+	blocking = FALSE
+	if(istype(I, /obj/item))
+		grabbed.attackby(I, attacker)
+	else
+		attacker.attack_hand(grabbed)//and now it's not our problems
+	blocking = TRUE
+	//change our block state depending on grab level
+	if(G.state < GRAB_NECK)
+		return //block remains active
+	else if(G.state < GRAB_AGGRESSIVE)
```

Comment:
```suggestion
	if(G.state >= GRAB_NECK)
		return //block remains active
	else if(G.state >= GRAB_AGGRESSIVE)
```
Checks were backwards. `G.state < GRAB_NECK` was catching the regular grabs and keeping the block up.